### ISSUE TITLE
Getting `unspecified-encoding` with Path().open("w")

### ISSUE THREAD

**Issue by yarnabrina** [PyCQA/pylint#5599]
I understood from #5017 that it is expected to be solved by version `2.11.1`, but I'm facing same issues even now. I do not get this with binary files (`"wb"`) though.

#### Code

```python3
"""Script to highlight misleading pylint warning."""

import pathlib

FILE = pathlib.Path("hello_world.txt")

with FILE.open("w") as FILE_STREAM:
    FILE_STREAM.write("hello world")

```

### Folder Structure

```zsh
❯ tree
.
└── hello_world.py

0 directories, 1 file
```

### Configuration

**None**

### Invocation Command

> pylint --enable=all hello_world.py

(Same happens without `--enable=all` as well)

### Pylint Output

```zsh
❯ pylint --enable=all hello_world.py
************* Module hello_world
hello_world.py:7:5: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)

------------------------------------------------------------------
Your code has been rated at 7.50/10 (previous run: 7.50/10, +0.00)


```

### Expected Output

**None**, with perfect score

### Pylint Version

```
❯ pylint --version
pylint 2.12.2
astroid 2.9.0
Python 3.10.0 (default, Nov 10 2021, 11:24:47) [Clang 12.0.0 ]
```

### OS Version

> macOS Monterey 12.0.1

**Comment by DanielNoord** [comment]
I believe this is expected behaviour. You're not specifying that the file should be openend in binary mode, so it's openend in text mode making an encoding "required". The issue you're referring to indicated that we did not recognise the binary mode parameter correctly. This has been resolved.

**Comment by yarnabrina** [comment]
@DanielNoord : I do not want to open it in binary mode at all. I want to open it in text mode only, and it is a completely valid python code. It should not raise any warning at all.

If it is an "expected issue", how exactly will you specify the encoding?

(And I know binary mode parameter is solved. Said so myself in the main thread already.)

**Comment by DanielNoord** [comment]
@yarnabrina Whether you open in binary or text mode is referred to as the `mode` in which you open. The encoding is the codec used to open the file. You can fix it (for example) by doing `open("w", encoding="utf-8")`.
Please also the official Python documentation which explains the difference between them [here](https://docs.python.org/3/library/functions.html#open).

Those docs also refer to as to why we have this check in the first place: default encodings differ per platform and therefore without specifying your encoding your code might fail or run differently on other platforms than the one you're currently working on.
